lib/max_subarray.py

```diff
@@ -2,11 +2,18 @@
 def max_sub_array(nums):
     """ Returns the max subarray of the given list of numbers.
         Returns 0 if  nums is None or an empty list.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n) where n is len(nums)
+        Space Complexity: O(1)
     """
     if nums == None:
         return 0
     if len(nums) == 0:
         return 0
-    pass
+
+    max_so_far = nums[0]
+    curr_max = nums[0]
+    for i in range(1, len(nums)):
+        curr_max = max(curr_max + nums[i], nums[i])
```

Comment:
✨ This is a really nice way to represent this calculation, which captures the underlying invariant that makes Kadane's algorithm work.